proposals/0279-multiple-trailing-closures.md

```diff
@@ -1,422 +1,369 @@
+
+For example, given this function:
 
 ```swift
-when(2 < 3,
-  then: { ... },
-  else: { ... }
-)
+func when<T>(_ condition: @autoclosure () -> Bool,
+             then: () -> T,
+             `else`: () -> T) -> T {
+  ...
+}
 ```
 
-Note that the new syntax also allows specifying closures for unnamed parameters by using an “empty” label placeholder `_:` for each, e.g.,
+The following call using the new syntax:
 
 ```swift
-func foo(_ fn1: () -> Void, _ fn2: () -> Void) {}
-
-foo {
-  _: { ... }
-  _: { ... }
+when(2 < 3) {
+  print("then")
+} else: {
+  print("else")
 }
 ```
 
-Is equivalent to:
+is equivalent to:
 
-```swift
-foo({ ... }, { ... })
+```
+`when(2 < 3, then: { print("then") }, else: { print("else") })`
 ```
 
-The same applies even when using the proposed syntax for a single trailing closure, which provides a trailing syntax for specifying 
-the argument label of a function that accepts a single closure parameter:
+It's important to note that the handling of default arguments in relation to trailing closures is maintained as-is.  For example:
 
 ```swift
-func foo(fn: () -> Void) {
-}
-
-foo {
-  _: { ... }
-}
+func foo(a: () -> Int = { 42 }, b: Any? = nil) {}
 
 foo {
-  fn: { ... }
+  42
 }
 ```
 
-Would be transformed into:
+Although trailing closure matches parameter for `a:` by type, existing trailing closure behavior would match trailing closure argument to parameter labeled as `b:`, which means that previous call to foo is equivalent to:
 
 ```swift
-foo({ ... })
-
-foo(fn: { ... })
+foo(b: { 42 })
 ```
 
-First call is treated as if being a single trailing closure `foo { ... }`, second call uses new syntax to preserve a label, both calls are accepted.
-
-Since all of the essential source information is preserved (locations of all labels and closure blocks) it would be possible for the type-checker to produce diagnostics for invalid code without any changes.
+Now let's add one more parameter to foo to see how this applies to the new multiple trailing closures syntax:
 
-## Source compatibility
-
-This is an additive proposal, which makes ill-formed syntax well-formed but otherwise does not affect existing code.
-
-## Effect on ABI stability
+```swift
+func foo(a: () -> Int = { 42 }, b: Any? = nil, c: () -> Void) {}
 
-This feature is implementable entirely in the parser, as a syntactic transformation on call expressions. It, therefore, has no impact on the ABI.
+foo {
+  42
+} c: {
+  ...
+}
+```
 
-## Effect on API resilience
+Since new type-checking rule dictates a backwards scan starting for the last (labeled) trailing closure before attempting to match an unlabeled argument, this call is equivalent to a following "old" syntax:
```

Comment:
* Add missing "the"

```suggestion
Since the new type-checking rule dictates a backwards scan starting for the last (labeled) trailing closure before attempting to match an unlabeled argument, this call is equivalent to a following "old" syntax:
```